app.rb

```diff
@@ -1,12 +1,143 @@
+require_relative './book'
 require_relative './person'
+require_relative './rental'
 require_relative './student'
 require_relative './teacher'
-require_relative './capitalize_decorator'
-require_relative './trimmer_decorator'
-
-person = Person.new(354, 'Frankestein')
-person.correct_name
-capitalized_person = CapitalizeDecorator.new(person)
-capitalized_person.correct_name
-capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
-p capitalized_trimmed_person.correct_name
+require_relative './views/main_screen'
+
+class App
+  def initialize
+    @screen = MainScreen.new
+    @person_list = []
+    @book_list = []
+    @rental_list = []
+  end
+
+  def run
+    @screen.initial
+    print '   Select a option number, then press [ENTER]: '
+    opc = gets.chomp.to_i
+
+    case opc
+    when 1
+      list_book
+    when 2
+      list_people
+    when 3
+      add_person
+    when 4
+      add_book
+    when 5
+      rent_a_book
+    when 6
+      rent_history
+    else
+      @screen.exit
+    end
+  end
+
+  # OPTION 1
+  def list_book
+    @screen.list_books(@book_list)
+    gets.chomp
+    run
+  end
+
+  # OPTION 2
+  def list_people
+    @screen.list_people(@person_list)
+    gets.chomp
+    run
+  end
+
+  # OPTION 3
+  def add_person
+    @screen.add_person
+    print '   Enter [1] for Student, or [2] for a teacher: '
```

Comment:
- [ ] Kindly make sure your application is able to add a new person to make it match all the project's requirements. Right now when trying to add a new person  the app raises the `ArgumentError`

| Error To Add a Student | Error To Add a Teacher |
| ------------------------------- | --------------------------------- |
| ![Screenshot from 2022-08-26 03-50-16](https://user-images.githubusercontent.com/83042742/186799318-8aa86001-a782-4219-877b-92ff5c3667e8.png) | ![Screenshot from 2022-08-26 03-52-22](https://user-images.githubusercontent.com/83042742/186799552-88c1fe5a-389d-4e37-bf9f-d2bc2cebf005.png) |

- [ ] Once you fix the above issue, make sure you also check the feature to `Rent a book` as right now it cannot be checked because impossible to add a person(i.e. the list of persons is empty)

Reply:
@tresorsawasawa There is a compatibility error with Rubocop and Ruby, when you initialize boolean variables Rubocop requires you to use ":" however, Ruby does not work, only with "=", believe me I have searched all day for a way to solve it and it does not exist